dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ForkingClusterInvoker.java

```diff
@@ -86,6 +86,9 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
                 URL consumerUrl = RpcContext.getServiceContext().getConsumerUrl();
                 executor.execute(() -> {
                     try {
+                        if (ref.size() > 0) {
```

Comment:
Very good, but the concurrency problem must be taken into account. Suppose that 10 threads execute this step and judge size=0. At this time, multiple elements will still be stored in the queue. The size judgment can use the atomic class, like the previous `final AtomicInteger count = new AtomicInteger();`

Reply:
I think that in the case of 10 threads judging at the same time, there should not be a limit, because the results are not obtained at this time and as many threads as possible need to initiate the access, so as to speed up the acquisition of results

Reply:
However, I have added a capacity limit to the queue, which prevents the memory usage caused by multiple threads getting the results at the same time

Reply:
Setting size to 1 can only ensure that the result of each call by multiple threads will not be stored when queue.offer is executed, but the actual call still occurs.
Of course, this is a probabilistic event under multithreading.

Reply:
> I think that in the case of 10 threads judging at the same time, there should not be a limit, because the results are not obtained at this time and as many threads as possible need to initiate the access, so as to speed up the acquisition of results

Yes, as stated above, concurrent calls cannot be limited because of the need to get the fastest results